src/Language/Spectacle/Exception/RuntimeException.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
+module Language.Spectacle.Exception.RuntimeException
+  ( RuntimeException (VariableException),
+    VariableException (CyclicReference),
+  )
+where
+
+import Control.Exception (Exception)
+import Type.Reflection (Typeable)
+
+-- -------------------------------------------------------------------------------------------------
+
+data RuntimeException where
+  VariableException :: VariableException -> RuntimeException
+  deriving (Exception, Show, Typeable)
+
+data VariableException where
+  CyclicReference :: [String] -> VariableException
+  deriving (Exception, Show, Typeable)
```

Comment:
Any reason these aren't `newtype`s?

Also, I recommend using `DerivingStrategies` to be more explicit about how things are derived.

Reply:
They'll both be extended in upcoming PRs, we still need exceptions for universal/existential quantification for example. I do think that deriving strategies would help with readability though.